### PR TITLE
fix file ordering

### PR DESCRIPTION
**Pull request by szhengac**
`get_optim_files` in `zero_to_fp32.py` does not sort files correctly: 1, 10, 2, 3, 4, 5, 6, 7, 8, 9, leading to the wrong merged pytorch checkpoint. Add natural ordering to correct it.

### CODE REVIEW

File: deepspeed/utils/zero_to_fp32.py [hunk 1]
```diff
@@ -52,7 +61,7 @@ def get_model_state_file(checkpoint_dir, zero_stage):
 
 def get_optim_files(checkpoint_dir):
     # XXX: need to test that this simple glob rule works for multi-node setup too
-    optim_files = sorted(glob.glob(os.path.join(checkpoint_dir, "*_optim_states.pt")))
+    optim_files = sorted(glob.glob(os.path.join(checkpoint_dir, "*_optim_states.pt")), key=natural_keys)
```
Comment (tjruwase) [inline deepspeed/utils/zero_to_fp32.py:64]: Can you please share unit test for this case?

Comment (jeffra) [reply deepspeed/utils/zero_to_fp32.py:69]: Unfortunately, our CI does not currently support multi-node testing :(

Maybe a unit test that calls `get_optim_files` directly and passes a `tmpdir` path where some empty files were created with the `*_optim_states.pt` naming style.

Comment (szhengac) [reply deepspeed/utils/zero_to_fp32.py:69]: Makes sense to me. I will prepare one.

Comment (szhengac) [reply deepspeed/utils/zero_to_fp32.py:69]: @jeffra @tjruwase unittest is added

**Comment by szhengac** [comment]
@tjruwase any update on this PR?

**Review by jeffra — APPROVED** [review]
Thank you @szhengac! Looks good to me.

**Pull request merged**